src/components/user/PersonalProfileEditForm.js

```diff
@@ -44,6 +55,23 @@ export default function PersonalProfileEditForm() {
       });
     } else {
       try {
+        // If there's a profile picture, upload it first
+        if (formData.profilePicture) {
+          const formDataImage = new FormData();
+          formDataImage.append('image', formData.profilePicture);
+
+          // Make the API call to upload the image
+          const imageResponse = await axiosInstance.post(
+            '/:userId/uploadProfile',
```

Comment:
This will actually make a call to `/:userId/uploadProfile` axios will assume that is the literal string you want to use:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/ecb32e8f-92fc-4ed4-bf6e-3885d15c1949)

To substitute a variable we can use Javascript Template Strings (the backticks \` \` ). You are also missing the `/users` part of the url (remember that `axiosInstance` will handle the base part of the url (<localhost:8000 or musicstore.onrender.com>/api/v1`) but we still need to provide the rest: 

```javascript
          const imageResponse = await axiosInstance.post(
            `/users/${user?._id}/uploadProfile`,
            formDataImage
          );
```

Reply:
Thank you , changed commit f6aad6f76ab1b16a05fd7af3c08e91400292675f